### ISSUE TITLE
Clarify and expand on component use

### ISSUE THREAD

**Issue by bob2517** [Active-CSS/active-css#21]
Offline, I've added a way to assign a component to an element as a parameter of create-element. That gives a more traditional way of setting up a custom element and doesn't look as messy as manually rendering components in a draw event. On doing this, I spotted that it looks like the lifecycle callback events are not going to work in nested components, so that needs fixing.

The docs are really weak in the area of using components. The whole topic is essentially very simple, but even I'm getting confused reading my docs.

These are just thoughts before I tackle the docs, but probably:

0) For each custom element doc example use create-element in "body:init{}," to attach the component, and remove the manual rendering from the docs. Either way is allowed but it just looks less messy using create-element.
1) We don't have any lifecycle callback examples at all. So set up an example for lifecycle callbacks in the concept examples section - shadow DOM component.
2) Set up an example for lifecycle callbacks - non-shadow DOM component.
3) Set up an example for nested component lifecycle callbacks - mixed component types.
4) Fix any lifecycle callback bugs found as you go.
5) Set up the components section so it's a menu.
6) Set up a component-driven development method.
6) Have separate pages for each of the parts of the traditional web components:
    a) Have an introduction page, covering what a component is.
    b) Have an "setting up events in components" page.
    c) Have a "using variables in components" page.
    d) Have a "setting up custom elements" page.
    e) Have a "using lifecycle callbacks" page.
    f) Have a component-driven development page.

I think this should be all that's needed for doing the definitive tutorial. Possibly. Probably not... there's stil some things that Active CSS can't handle easily.

**Comment by bob2517** [comment]
PS. If anyone is reading this and needs the lifecycle callback bug fixing urgently prior to 2.4.0, drop me a line via support so I know and I'll push it up when it's done. Otherwise, it will go live with 2.4.0.

**Comment by bob2517** [comment]
Found the issue with the lifecycle callbacks. The issue is that the create-element events are only running in the document scope. Hmmm... we do know the component at the moment of creation if there is a component referenced in the create-element command, which can be done now (offline) so that's ok. But what if it was rendered outside of create-element manually? If it isn't present, it could be gotten from looking at which scope the element is placed. Should be ok I think...

**Comment by bob2517** [comment]
I got a feeling this bug extends into nested regular non-private events from looking at the code. Need to test though to confirm, which I'll do shortly. If anyone has found any errors in this area, let me know via support on the website.

**Comment by bob2517** [comment]
Actually no, nested non-private events do bubble ok. That's fine. Phew! Hope I'm not speaking too soon...

**Comment by bob2517** [comment]
Lifecycle callback fix almost done - just tying in a few things. And I just realised the attribute updating examples I've got on the docs site are copied from the MDN lifecycle callback component examples which I rewrote without actually using lifecycle callbacks. Doh. I'm just going to convert those. That should give a clear comparison of lifecycle callbacks and demonstrate that Active CSS is easier to read than native JS at the same time.
